doc/HttpFiltering.md

```diff
@@ -0,0 +1,55 @@
+## HTTP Filtering
+
+With the release of sync-android version TBD a new HTTP  Filtering API was introduced.
+HTTP filtering allows the developer to modify the HTTP Requests and Responses during
+the replication process.
+
+In order to make changes to a HTTP request, you need to implement the interface
+`HTTPConnectionRequestFilter`, for HTTP Responses, you implement the interface
+`HTTPConnectionResponseFilter`. A Class may implement both interface without causing
+problems, an example of this is the `CookieFilter` class.
+
+In order to add an HttpFilter to a replication, you call the `addResponseFilters` or
+`addRequestFilters` on the `ReplicatorBuilder` class.
+
+For example, this is how you would add an instance of `CookieFilter` to a pull replication.
+
+```java
+import com.cloudant.sync.replication.Replicator
+import com.cloudant.sync.replication.ReplicatorBuilder
+import com.cloudant.sync.datastore.Datastore
+import com.cloudant.sync.http.CookieFilter
+
+
+Datastore ds =  manager.open("my_datastore");
+URI cloudant = new URI("https://username.cloudant.com");
+CookieFilter filter = new CookieFilter("username","password");
+
+
+
+Replicator replicator = ReplicatorBuilder.pull()
+							.from(cloudant)
+							.to(ds)
+							.addRequestFilters(filter)
+							.addResponseFilters(filter)
+							.build();
+
+replicator.start();
+
+```
+
+Filters can be used to implement your own authentication schemes, for example OAuth, or
+provide custom headers so you can perform your own analysis on usage. For an example
+of how to implement a filter, see  the `CookieFilter` class.
+
+
+## Things Not to do
+
+The `HttpConnectionFilterContext` object provides access to the underlying
+`HttpConnection` and `HttpURlConnection` classes. This allows you to change
+settings and interact with the connection that will cause errors in the replicator.
```

Comment:
"interact with the connection in ways would could potentially cause errors in the replicator"